aqueducthq/aqueduct
# [1/2] Eng-2503: Refactor Conda Connection Status for more generic purposes

## Describe your changes and why you are making these changes
This PR is part of the ENG-2503 which enables Lambda to connect asynchronously. The PR specifically refactors functions under `lib/execution_environment/integration.go` and move them under  `lib/execution_state/utils.go`. 
Now the `updateOnFailure()` function takes `IntegrationConfig ` instead of `condaPath` so that it can update failure on different engine connection. 

Testing:
1. Try to initialize a Conda Environment, UI changes from "Running" to "Succeed"
2. Try to initialize a bad Conda Environment(Conda build not installed), UI changes from " Running" to "Error"
Since this part doesn't pertain to actual Conda initialization, I only test whether the each status is passed through as expected. 

## Related issue number (if any)

## Loom demo (if any)

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)
